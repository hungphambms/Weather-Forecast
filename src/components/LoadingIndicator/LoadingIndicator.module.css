.loading {
    font-size: 16px;
    margin: 24px 0;
    text-align: center;

    font-weight: 500;
    text-transform: uppercase;
}

.loading img {
    font-size: 21px;
    margin-right: 16px;
}