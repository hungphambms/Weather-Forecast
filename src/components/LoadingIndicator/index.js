import PropTypes from "prop-types";

import styles from "./LoadingIndicator.module.css";

import imgLoader from "./loader.gif";

function LoadingIndicator({ isLoading, children }) {
  return isLoading ? (
    <div className={styles.loading}>
      <img src={imgLoader} alt="loader" />
      Loading
    </div>
  ) : (
    children
  );
}

LoadingIndicator.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  children: PropTypes.object.isRequired,
};

export default LoadingIndicator;
