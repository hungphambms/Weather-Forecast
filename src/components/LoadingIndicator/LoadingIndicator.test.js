/* eslint-disable jsx-a11y/no-redundant-roles */
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

import LoadingIndicator from "./index";

afterEach(cleanup);

describe("LoadingIndicator component rendering", () => {
  it("renders LoadingIndicator component without crashing", () => {
    const component = renderer.create(
      <LoadingIndicator isLoading>
        <h1>Sample Children Content</h1>
      </LoadingIndicator>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render children", () => {
    const component = render(
      <LoadingIndicator isLoading={false}>
        <h1 role="heading">Sample Children Content</h1>
      </LoadingIndicator>
    );

    const headingContainer = component.getByRole("heading");
    expect(headingContainer.textContent).toBe("Sample Children Content");
  });
});
