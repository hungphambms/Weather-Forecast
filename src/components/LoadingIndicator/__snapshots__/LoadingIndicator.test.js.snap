// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator component rendering renders LoadingIndicator component without crashing 1`] = `
<div
  className="loading"
>
  <img
    alt="loader"
    src="loader.gif"
  />
  Loading
</div>
`;
